assets/src/date/getTimeSensitiveDisplayDate.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { format } from '@wordpress/date';
+
+/**
+ * External dependencies
+ */
+import moment from 'moment-timezone';
+
+/**
+ * Internal dependencies
+ */
+import {
+  DEFAULT_DATE_FORMATTING,
+  getDateFormattedFromWordPressToMoment,
+  getTimeFromNow,
+  isToday,
+  isYesterday,
+} from './';
+
+// Returns hours since date if date passed in matches today
+// Returns yesterday if date passed in is yesterday
+// Returns formatted date + time by default
+/**
+ * @summary                        Formats a date to display relative to time passed since date.
+ * If date to display is < 1 day ago it will display rounded time since date using timezone.
+ * If date to display matches yesterday's date it will display "yesterday".
+ * Otherwise date will come back formatted by dateFormatting.dateFormat (no time).
+ * @param {Date} date              Date to format according to how much time or how many days have passed since date
+ * If date is not an instance of moment when passed in it will create a moment from it.
+ * @param {Object} dateFormatting  Object responsible for relevant date formatting.
+ * Should contain dateFormat, timezone, gmtOffset, and timeFormat - all strings.
+ * @return {string}                Displayable relative date string
+ */
+
+export function getTimeSensitiveDisplayDate(
```

Comment:
Time-sensitive might not be the best naming here. I think of time-sensitive as being urgent.

Why  not just `getDisplayDate`?

Reply:
I had mixed feelings about this name. I went with timeSensitive because the date renders conditionally depending on how long it's been so it would be misleading to call it 'getDisplayDate' because if you're expecting a consistent mm-dd-yyyy and you get 'yesteday' that seems odd. 

Since timeSensitive feels urgent, I will update this to be `getRelativeDisplayDate` since we are displaying the date as it is relative to today.